docs/spec/nmt-wrapper.md

```diff
@@ -0,0 +1,83 @@
+# Namespaced Merkle Tree Wrapper
+## Abstract
+In Celestia, block transactions are grouped into identical-size shares and arranged in a `k` by `k` matrix, called [original data square](https://github.com/celestiaorg/celestia-app/blob/specs-staging/specs/src/specs/data_structures.md#arranging-available-data-into-shares).
+The rows and columns of the original data square are then extended using a 2D Reed-Solomon coding scheme.
+The [extended version](https://github.com/celestiaorg/celestia-app/blob/specs-staging/specs/src/specs/data_structures.md#2d-reed-solomon-encoding-scheme) is called extended data square, that is a `2k` by `2k` square consisting of `4` quadrants namely, `Q0`, `Q1`, `Q2`, and `Q3`.
+Figure 1 provides an illustration of a sample data square and its extended version.
+`Q0` corresponds to the original data square.
+`Q1` and `Q2` represent the horizontal and vertical extensions of `Q0`, respectively. 
+`Q3` is the extension of `Q2` in the horizontal direction.
+```markdown
+  k         k
+ -------   -------
+|        |        |
+|   Q0 → |   Q1   | k
+|   ↓    |        |
+-------  -------
+|        |        |
+|   Q2 → |   Q3   | k
+|        |        |
+ -------   -------
+```
+Figure 1.
+
+Each row and column of the extended data square is modeled by a [Namespace Merkle Tree](https://github.com/celestiaorg/nmt/blob/master/spec/nmt.md).
+NMTs require the data items they represent to be namespaced, which means that the shares within each row or column of the extended data square must be namespaced before being added to the NMT.
+This is where the [Namespaced MerkleTree Wrapper](https://github.com/celestiaorg/celestia-app/blob/main/pkg/wrapper/nmt_wrapper.go)  comes into play.
```

Comment:
[question] the linked file only has a `ErasuredNamespacedMerkleTree`, is that what this is referring to? If so, is this alright:

```suggestion
This is where the [ErasuredNamespacedMerkleTree](https://github.com/celestiaorg/celestia-app/blob/main/pkg/wrapper/nmt_wrapper.go) comes into play.
```

Reply:
Yes, the NMT wrapper refers to the `ErasuredNamespacedMerkleTree` struct, I revised the text to make it clear.  Please see https://github.com/celestiaorg/celestia-app/pull/1442/commits/22e737c961c551e85166cfd9ecb455d248c5d8c4.